rpcserver.go

```diff
@@ -2087,17 +2087,22 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
 		return nil, fmt.Errorf("cannot open channel to self")
 	}
 
-	// Calculate an appropriate fee rate for this transaction.
-	feeRate, err := lnrpc.CalculateFeeRate(
-		uint64(in.SatPerByte), in.SatPerVbyte, // nolint:staticcheck
-		uint32(in.TargetConf), r.server.cc.FeeEstimator,
-	)
-	if err != nil {
-		return nil, err
-	}
+	var feeRate chainfee.SatPerKWeight
 
-	rpcsLog.Debugf("[openchannel]: using fee of %v sat/kw for funding tx",
-		int64(feeRate))
+	// Skip estimating fee rate for PSBT funding.
+	if in.FundingShim == nil || in.FundingShim.GetPsbtShim() == nil {
```

Comment:
Why should the fee rate be skipped? IIUC, you can provide something with empty or partial inputs to have coin selection figure out the rest (change addrs, etc too)

Reply:
The fee estimation needs to be skipped because of this check,
https://github.com/lightningnetwork/lnd/blob/a3bf2c70e5aabf01c5db536f4cb90d2080c65b05/rpcserver.go#L1879-L1882

> This change seems unrelated to the PR - I'm missing the motivation here?

Because we enforce fee params check in `CalculateFeeRate`, if we don't pass them, it will fail there. And if we pass them, it will fail for PSBT funding.